script.js

```diff
@@ -1,12 +1,164 @@
+  let nameA = a.name.toLowerCase();
+  let nameB = b.name.toLowerCase();
+  if (nameA < nameB) return -1;
+  if (nameA > nameB) return 1;
+  return 0;
+});
+// build the shows dropdown list
+allShows.forEach((show) => {
+  // let URL = "http://api.tvmaze.com/shows/" + show.id + "/episodes";
+  let showsDropDownElem = document.getElementById("shows-dropdown");
+  let showsDropDownOptions = document.createElement("option");
+  showsDropDownOptions.textContent = show.name;
+  showsDropDownOptions.value = show.name;
+  showsDropDownElem.appendChild(showsDropDownOptions);
+});
+
+//getting all the shows
+async function displayShowsEpisodes(e) {
+  let API_URL = "";
+
+  allShows.forEach((show) => {
+    if (e.target.value === show.name) {
+      API_URL = "http://api.tvmaze.com/shows/" + show.id + "/episodes";
+    }
+  });
+
+  const fetchResult = await fetch(API_URL);
+
+  const data = await fetchResult.json();
+
+  makePageForEpisodes(data);
+}
+
+// this function pads the season and episode numbers with a 0 when they are only 1 digit
+function zeroPadder(inputNumber) {
+  return inputNumber < 10 ? "0" + inputNumber : inputNumber;
+}
+// this function builds the page
 function makePageForEpisodes(episodeList) {
+  const spanEl = document.getElementsByTagName("span")[0];
+  spanEl.textContent = `${episodeList.length}/${episodeList.length} showing`;
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  const dropDownElem = document.getElementById("drop-down");
+
+  const allElem = document.createElement("option");
+  allElem.value = "Main-page";
+  allElem.textContent = "Main-page";
+
+  dropDownElem.appendChild(allElem);
+
+  episodeList.forEach((episode) => {
+    const { name, image, season, number, summary } = episode;
+    const episodeEl = document.createElement("div");
+    episodeEl.classList.add("episode-card");
+
+    episodeEl.innerHTML = `
+      <div class="season-title">
+          <h1>S${zeroPadder(season)}-E${zeroPadder(number)}</h1>
+          <h2>${name}</h2>
+      </div>
+      
+      <img src="${image ? image.medium : ""}" alt="${name}">
+      <div class="summary-text">
+        <h3>${summary}</h3>
+      </div>`;
+
+    const dropDownOptions = document.createElement("option");
+    dropDownOptions.textContent = `S${zeroPadder(season)}-E${zeroPadder(
+      number
+    )}:${name}`;
+    dropDownOptions.value = name;
+
+    dropDownElem.appendChild(dropDownOptions);
+    rootElem.appendChild(episodeEl);
+  });
 }
 
-window.onload = setup;
+// The search feature
+function searchResults(e) {
+  const spanEl = document.querySelector("span");
+
+  let query = e.target.value.toLowerCase().trim();
+
+  const movieTitleEls = document.querySelectorAll(".season-title");
+  const movieSummaryEls = document.querySelectorAll(".summary-text");
+  const episodeCardEls = document.querySelectorAll(".episode-card");
+  query.split(" ").map((character) => {
```

Comment:
This is quite labour intensive to show/display each episode card. Wouldn't it better to filter out the episodes that you dont want and rebuild the cards using ``makePageForEpisodes`` using the filtered episode list?

Reply:
Again , not sure what you mean .